api/v1alpha4/virtualmachinesnapshot_types.go

```diff
@@ -0,0 +1,142 @@
+// © Broadcom. All Rights Reserved.
+// The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha4
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VirtualMachineSnapshotSpec defines the desired state of VirtualMachineSnapshot.
+type VirtualMachineSnapshotSpec struct {
+	// +optional
+	// +kubebuilder:default=false
+	//
+	// Memory represents whether the snapshot includes the VM's
+	// memory. If true, a dump of the internal state of the virtual
+	// machine (a memory dump) is included in the snapshot. Memory
+	// snapshots consume time and resources and thus, take longer to
+	// create.
+	// The virtual machine must support this capability.
+	// When set to false, the power state of the snapshot is set to
+	// false.
+	// For a VM in suspended state, memory is always included
+	// in the snashot.
+	Memory *bool `json:"memory,omitempty"`
+
+	// +optional
+	// +kubebuilder:default=false
```

Comment:
This kubebuilder validation (default=false) will err out for non-boolean types..